awx/wsgi.py

```diff
@@ -29,5 +35,33 @@
         logger.error("Missing or incorrect metadata for Tower version.  Ensure Tower was installed using the setup playbook.")
         raise Exception("Missing or incorrect metadata for Tower version.  Ensure Tower was installed using the setup playbook.")
 
+
+if django.__version__ != '1.11.7':
+    raise RuntimeError("Django version other than 1.11.7 detected {}. \
+            Monkey Patch to support short-circuit Django Middelware \
+            is known to work for Django 1.11.7 and may not work with other, \
+            even minor, versions.".format(django.__version__))
+
+
+if settings.MIDDLEWARE:
+    raise RuntimeError("MIDDLEWARE setting detected. \
+            The 'migration in progress' view feature short-circuits OLD Django \
+            MIDDLEWARE_CLASSES behavior. With the new Django MIDDLEWARE beahvior \
+            it's possible to short-ciruit the middleware onion through supported \
+            middleware mechanisms. The monkey patch wrapper below should be removed.")
+
+
+def _wrapper_legacy_get_response(self, request):
+    # short-circuit middleware
+    if getattr(resolve(request.path), 'url_name', '') == 'migrations_notran':
+        return self._get_response(request)
+    # fall through to middle-ware
+    else:
+        return self._real_legacy_get_response(request)
+
+
+BaseHandler._real_legacy_get_response = BaseHandler._legacy_get_response
+BaseHandler._legacy_get_response = _wrapper_legacy_get_response
```

Comment:
Do we _need_ to monkey patch, or does Django have any sort of config that allows us to override the handler with a subclass where we could define our own `_legacy_get_response`?